test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -482,15 +484,28 @@ def _dist_train(self, wrap_fsdp: bool, state_dict_type: str = ""):
             blank_model = FSDP(Model(True).cuda())
             _zero_model(blank_model)
             state_dict = self._state_dict(model, state_dict_type)
+            if move_to_cpu:
+                for key in list(state_dict.keys()):
+                    tensor = state_dict[key]
+                    if isinstance(tensor, torch.Tensor):
+                        state_dict[key] = tensor.cpu()
+                    else:
+                        shards = tensor.local_shards()
+                        if shards:
+                            shards[0].tensor = shards[0].tensor.cpu()
+
             self._load_state_dict(blank_model, state_dict_type, state_dict)
             return get_full_params(blank_model)
         else:
             return list(model.parameters())
 
     @skip_if_lt_x_gpu(2)
     @parametrize("state_dict_type", _SUPPORTED_STATE_DICT_IMPLS)
-    def test_state_dict_save_load_flow(self, state_dict_type):
-        fsdp_params = self._dist_train(wrap_fsdp=True, state_dict_type=state_dict_type)
+    @parametrize("move_to_cpu", [True, False])
+    def test_state_dict_save_load_flow(self, state_dict_type, move_to_cpu):
+        fsdp_params = self._dist_train(
```

Comment:
nit: instead of parametrize, use 

```
for move_to_cpu in [True, False]:
    with self.subTest(move_to_cpu=move_to_cpu):
        # test code
```

as this reuses the processes instead of re-spawning them